data-generation/src/main/scala/thesis/LandyConsumer.scala

```diff
@@ -0,0 +1,72 @@
+package thesis
+
+import org.apache.spark.SparkContext
+import org.apache.spark.graphx.{Edge, Graph}
+import org.apache.spark.rdd.RDD
+import thesis.Action.{CREATE, UPDATE}
+import utils.{LogUtils, UtilsUtils}
+
+import java.time.Instant
+import scala.collection.mutable
+
+object LandyConsumer {
+  def createEdge(log: LogTSV, validTo: Instant): Edge[LandyEdgePayload] = {
+    log.entity match {
+      case Entity.VERTEX(_) => throw new IllegalStateException("This should not be called on a log with vertices")
+      case Entity.EDGE(id, srcId, dstId) => {
+        val payload = LandyEdgePayload(id=id, validFrom = log.timestamp, validTo = validTo, attributes = log.attributes)
+        Edge(srcId, dstId, payload)
+      }
+    }
+  }
+
+  def createVertex(log: LogTSV, validTo: Instant): (Long, LandyVertexPayload) = {
+    log.entity match {
+      case Entity.VERTEX(objId) => {
+        val payload = LandyVertexPayload(id=objId, validFrom = log.timestamp, validTo = validTo, attributes = log.attributes)
+        (UtilsUtils.uuid, payload)
+      }
+      case Entity.EDGE(_, _, _) => throw new IllegalStateException("This should not be called on a log with edges")
+    }
+  }
+
+  def generateEdges(logs: Seq[LogTSV]): Seq[Edge[LandyEdgePayload]] = {
+    val backwardOrderedLogs = logs.sortWith((log1, log2) => log1.timestamp.isBefore(log2.timestamp))
+    val aVeryBigDate = Instant.parse("9999-01-01T00:00:00.000Z")
+    var lastTimestamp = aVeryBigDate
```

Comment:
før så likte jeg at scala hadde `var` og `val` for mutable og immutable. Men nå synes jeg de er litt for like hverandre. Jeg stusset på denne linjen helt til jeg til slutt så at det var en `var`

Reply:
Ja enig. jeg kan lage issue på scala githuben å bytte til const og let jeg;)

Reply:
støttes !